tensorflow/python/keras/layers/normalization.py

```diff
@@ -736,7 +736,9 @@ def call(self, inputs, training=None):
     if self.virtual_batch_size is not None:
       # Virtual batches (aka ghost batches) can be simulated by reshaping the
       # Tensor and reusing the existing batch norm implementation
-      original_shape = [-1] + inputs.shape.as_list()[1:]
+      original_shape = [
+          d if d is not None else -1 for d in inputs.shape.as_list()]
```

Comment:
This is not super safe as it can generate more than one -1.

Isn't it better to use tf.shape instead?